Sources/W3CTraceContext/TraceContext.swift

```diff
@@ -44,3 +44,37 @@ public struct TraceContext: Equatable {
         self.state = TraceState(rawValue: stateRawValue) ?? TraceState([])
     }
 }
+
+extension TraceContext {
+    /// Replace the parent-id of the current `traceParent` with a newly generated one, using the system random number generator.
+    public mutating func regenerateParentID() {
```

Comment:
I don't understand this one -- why like that?

Isn't it normally: 

- parent id
- my id

and changing the parent actually means "now there's a new span, a child of me", thus:

- parent = my id
- my id = new id

?

Reply:
What did I miss / don't remember from the spec? Could you share a pointer?

Reply:
I agree that it feels strange that it's called `parentID`, but that's how I understand the spec:

> This is the ID of this request as known by the caller (in some tracing systems, this is known as the span-id, where a span is the execution of a client request)
> https://www.w3.org/TR/trace-context/#parent-id